File: BOJ/2170/2170_선긋기.py
import sys

input = sys.stdin.readline
lines = []

for i in range(int(input())):
    lines.append(list(map(int, input().split())))

lines.sort(key = lambda x: x[0])

check = False
start, end = 0, 0
ans = 0

for line in lines:
    if not check:
        start = line[0]
        end = line[1]
        check = True
        continue

    if line[0] <= end:
        end = max(end, line[1])

    else:
        ans += end - start
        start = line[0]
        end = line[1]
        
ans += end - start
print(ans)

File: BOJ/2170/2170_선긋기.js
// 입출력
const stdin = (require('fs').readFileSync('/dev/stdin').toString()).split('\n')
const input = (() => {
  let l = 0;
  return () => stdin[l++]
})()


const N = Number(input())
const lines = [];
let ans = 0, check = false, start = 0, end = 0;

// 로직
for (let i=0; i<N; i++) {
  lines.push(input().split(' ').map(Number))
}

lines.sort((a, b) => a[0] - b[0])

for (let line of lines) {
  if (!check) {
    start = line[0];
    end = line[1];
    check = true;
    continue
  }
  
  if (line[0] <= end) {
    end = Math.max(line[1], end)
  }
  
  else {
    ans += end - start;
    start = line[0];
    end = line[1];
  }
}
ans += end - start
console.log(ans)